arkouda/join.py

```diff
@@ -244,7 +250,11 @@ def inner_join(
             # Gather right whereargs
             rightWhere = whereargs[1][byRight.permutation][ranges]
             # Expand left whereargs
-            leftWhere = broadcast(fullSegs, whereargs[0][keep], ranges.size)
+            keep_where = whereargs[0][keep]
+            if isinstance(keep_where, Categorical):
+                leftWhere = broadcast(fullSegs, keep_where.codes, ranges.size)
+            else:
+                leftWhere = broadcast(fullSegs, keep_where, ranges.size)
```

Comment:
could we make it so `broadcast` accepts cats too and just operates on the `codes`? That way we don't have to change this at all? I def feel like broadcasting on categoricals is useful behavior